Coalescing the tables for platforms supported by Flutter deployment

sfshaza2 opened the pull request:
Staged:  https://sz-flutter-2.web.app/development/tools/sdk/release-notes/supported-platforms

Fixes https://github.com/flutter/website/issues/8365

@cbracken @timsneath @jmagman

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
+| web - Safari/Catalina  |     ✓     |             |             |
+| web - Edge 1.2.0       |     ✓     |             |             |
+| Windows 10             |     ✓     |             |             |
+| Windows 11          |(aspirational)|      ✓      |             |
```
Can we order consistently from newest to oldest? (Win10 and 11 should be flipped)...

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
+| Linux Debian 11        |           |      ✓      |             |
+| Linux Debian 10        |     ✓     |             |             |
+| Linux Debian 9 & below |           |      ✓      |             |
+| Linux Ubuntu 22.04 |(aspirational) |      ✓      |             |
```
is "in progress" a better term than "aspirational" here and for Win11?

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
+|Platform version        | Supported | Best effort | Unsupported |
+|------------------------|-----------|-------------|-------------|
+| Android SDK 19-30<sup>1</sup> | ✓  |             |             |
+| Android SDK 16-18      |           |      ✓      |             |
```
Not that it matters hugely, but the checkmarks aren't centered horizontally in the output.

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
+1. **Best-effort** platforms, supported through community
    testing, are platforms we believe we support through
    coding practices and ad-hoc testing,
    but rely on the community for testing.
```
This last clause feels a little odd -- we believe we support it through... testing, but we rely on the community for testing. I understand the point, but perhaps it's better to just say what we do offer, e.g. 

"platforms we intend to work through coding practices, but that we only test on an ad-hoc basis"

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
    but rely on the community for testing.
-1. Unsupported platforms, which are platforms that
+1. **Unsupported** platforms, which are platforms that
    might work, but that the development team
```
In fact, some of these platforms (e.g. Windows Vista / 32-bit) are those we know *cannot* work. Should we change this text to simply say, "platforms that the development team does not test or support"?

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
+| macOS High Sierra (10.13) & below | |            |      ✓      |
+| web - Chrome 84        |     ✓     |             |             |
+| web - Firefox 72.0     |     ✓     |             |             |
+| web - Safari/Catalina  |     ✓     |             |             |
```
This is an odd version number? There were many versions of Safari shipped on macOS Catalina, weren't there?

jmagman replied:
@yjbanov I don't know where Safari versioning enforcement lives, do you know?

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,75 @@ short-title: Supported platforms
+| web - Chrome 84        |     ✓     |             |             |
+| web - Firefox 72.0     |     ✓     |             |             |
+| web - Safari/Catalina  |     ✓     |             |             |
+| web - Edge 1.2.0       |     ✓     |             |             |
```
I don't recognize this numbering convention for Edge (perhaps it was "old" Edge?) 

Edge today embeds Chromium, and for example the current version shows as Microsoft Edge Version 111.0.1661.44 (Official build) (64-bit) on my machine. @yjbanov should this say Edge 84? What do we actually test?

sfshaza2 replied:
I've changed it to 111.0 but would appreciate your input @yjbanov.

timsneath reviewed (commented):
Added a bunch of comments for review. Some of them are not directly related to your changes, but it's worth taking a look at the promises made here while we're overhauling the page.

reidbaker commented:
Would you be willing to incorporate the changes from https://github.com/flutter/website/pull/8413 into this PR?

sfshaza2 commented:
Incorporated the changes from PR https://github.com/flutter/website/pull/8413.
@reidbaker, your chart left off Android SDK 20. I added it back in as supported. Please confirm.

@jmagman, @timsneath questioned the web - Safari/Catalina entry. Can you confirm or delegate?

Thanks, all! I've refreshed the staged version (listed at the top of this page).

timsneath reviewed (approved):
This is looking really good!

jmagman reviewed (approved):
iOS and macOS parts LGTM
Not sure about the Safari question, @yjbanov should know.

timsneath commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,69 @@ short-title: Supported platforms
+| web - Chrome 84          |<center>✓</center>| | |
+| web - Firefox 72.0       |<center>✓</center>| | |
+| web - Safari/Catalina    |<center>✓</center>| | |
+| web - Edge 111.0         |<center>✓</center>|           |        |
```
Sorry, I think I miscommunicated here. The version of Edge on my machine is 111.0 (that's the very latest version available). We support a much broader range of Edge versions, which should be the same as the range of Chrome versions (e.g. 84 and above), since they share the same engine and version numbering system.

cbracken reviewed (approved):
Looks great for desktop!

ditman commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,69 @@ short-title: Supported platforms
+| macOS Monterey (12) & above|<center>✓</center>| | |
+| macOS Mojave (10.14) - Big Sur (11) | |<center>✓</center>| |
+| macOS High Sierra (10.13) & below | | |<center>✓</center>|
+| web - Chrome 84          |<center>✓</center>| | |
+| web - Firefox 72.0       |<center>✓</center>| | |
+| web - Safari/Catalina    |<center>✓</center>| | |
+| web - Edge 111.0         |<center>✓</center>|           |        |
```
```suggestion
| web - Chrome |<center>latest 2</center>|<center>96+</center>| |
| web - Firefox |<center>latest 2</center>|<center>99+</center>| |
| web - Safari |<center>latest 2</center>| | |
| web - Edge | |<center>96+</center>| |
```

ditman replied:
/cc @timsneath I think web support is clearer now.

ditman reviewed (approved):
After some discussion, I left a suggestion that represents web support: "last 2" versions of modern browsers, but as a best effort, we normally fix issues in browsers that are older than "last 2". Those are indicated as "best effort".

loic-sharma commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,69 @@ short-title: Supported platforms
-   doesn't directly test or support.
+1. **Supported**<br>
+   Google-tested platforms that
+   are automatically tested at every commit
```
Should "at" be "on" here?

```suggestion
   are automatically tested on every commit
```

timsneath commented:
Thank you for wrangling this all, @sfshaza2! I think this page is significantly better. @itsjustkevin I wonder if we should review this page quarterly as a prerequisite to shipping a stable release?

itsjustkevin commented:
@timsneath sounds like a reasonable checklist item.  Do we make any guarantees on beta support currently?

sfshaza2 commented:
OK! Please give this another looksie. I'm sad to lose the sexy checkmarks, but I've followed the suggestion from @ditman for listing the web versions and, in fact, continued it for all platforms.

So, [here's the staged version, as of a week (or so) ago](https://sz-flutter-2.web.app/development/tools/sdk/release-notes/supported-platforms).

[And here is the post-@ditman version](https://sz-flutter.web.app/development/tools/sdk/release-notes/supported-platforms).

And, yes, @timsneath and @itsjustkevin, I definitely think this page should be revisited on every big release.

ditman commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,56 @@ short-title: Supported platforms
+| web - Firefox  |<center>latest 2 releases</center>|<center>99+</center>| |
+| web - Safari   |<center>latest 2 releases</center>|<center>| |
+| web - Edge     | |<center>96+</center>| |
+| Windows        | |<center>10</center>|<center>7 & 8</center>|<center>Vista-, any 32-bit</center>|
```
```suggestion
| Windows        |<center>10</center>|<center>7 & 8</center>|<center>Vista-, any 32-bit</center>|
```

Remove empty "first column" in the Windows row.

ditman commented on src/development/tools/sdk/release-notes/supported-platforms.md:
```diff
@@ -4,107 +4,56 @@ short-title: Supported platforms
 
-[iOS 10 & below and `arm7v` 32-bit iOS]: {{site.url}}/go/rfc-32-bit-ios-unsupported
+{{site.alert.note}}
+  A `plus` sign (`+`)` in the table indicates "and higher releases".
+  A `minus` (`-`) sign directly following a version (with no spaces)
+  indicates "and lower releases".
```
```suggestion
  A `plus` sign (`+`) in the table indicates "and higher releases".
  A `minus` (`-`) sign directly following a version (with no spaces)
  indicates "and lower releases".
```

Removed spurious backtick that was breaking formatting.

ditman reviewed (approved):
LGTM!

parlough reviewed (approved):
Looks good to me with all of those confirmations. Thanks everyone!

I'll likely follow up some follow-up infrastructure issues.

ditman commented:
🎉
